File: src/main/kotlin/edu/robocode/service/models/Battle.kt
package edu.robocode.service.models


import robocode.control.BattlefieldSpecification
import java.util.*

data class BattleSpecification(val numberOfRounds: Int, val inactivityTime: Long, val gunCoolingRate: Double, val robots: List<String>, val battlefieldSpecification: BattlefieldSpecification)
data class Battle(val id: UUID, val name: String, val started: Boolean, val specification: BattleSpecification)
data class NewBattle(val name: String, val specification: BattleSpecification)

File: client/src/app/models/battle.ts
import { Robot } from './robot';
export interface Battle {
  id: string;
  name: string;
  specification: BattleSpecification;
  started: boolean;
}

export interface BattleSpecification {
  numberOfRounds: number;
  inactivityTime: number;
  gunCoolingRate: number;
  robots: Robot[];
  battlefieldSpecification: BattlefieldSpecification;
}

export interface NewBattle {
  name: string;
  specification: NewBattleSpecification;
}

export interface NewBattleSpecification {
  numberOfRounds: number;
  inactivityTime: number;
  gunCoolingRate: number;
  robots: string[];
  battlefieldSpecification: BattlefieldSpecification;
}

export interface BattlefieldSpecification {
  height: number;
  width: number;
}


